### PR TITLE
Make device log streaming scale across multiple forge apps

### PR DESCRIPTION
**Pull request by hardillb**
part of https://github.com/FlowFuse/flowfuse/issues/3332
## Description

Move to subscribing to the device logs via MQTT in the browser

## Related Issue(s)

https://github.com/FlowFuse/flowfuse/issues/3332

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/routes/api/device.js [hunk 1]
```diff
@@ -592,6 +592,39 @@ module.exports = async function (app) {
...
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const team = await app.db.models.Team.byId(request.device.TeamId)
+        setTimeout(() => {
+            app.comms.devices.sendCommand(team.hashid, request.device.hashid, 'startLog', '')
+        }, 1000)
```
Comment (hardillb) [inline forge/routes/api/device.js:624]: This is to allow time for the client to connect before sending the log history

**Review by knolleary — COMMENTED** [review]
The websocket endpoint still exists for the frontend to connect to - if we don't need it, lets strip it out - along with the `streamLogs` stuff in `comms/devices.js`.

This still relies on the device logs being received by the platform just so it can decide whether to tell the device to keep logging or not. I'd love to find a way to avoid that.

Will have a ponder.

**Comment by hardillb** [comment]
>The websocket endpoint still exists for the frontend to connect to - if we don't need it, lets strip it out - along with the streamLogs stuff in comms/devices.js

I left it in as I didn't know if it was considered part of the v1 API

File: forge/routes/api/device.js [hunk 2]
```diff
@@ -624,13 +624,37 @@ module.exports = async function (app) {
...
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
         const team = await app.db.models.Team.byId(request.device.TeamId)
-        app.comms.devices.streamLogs(team.hashid, request.device.hashid, connection.socket)
+        setTimeout(() => {
+            app.comms.devices.sendCommand(team.hashid, request.device.hashid, 'startLog', '')
```
Comment (knolleary) [inline forge/routes/api/device.js:655]:
```suggestion
            app.comms.devices.sendCommand(team.hashid, request.device.hashid, 'startLog', '')
            app.log.info(`Enable device logging ${parts[1]}`)
```

File: forge/comms/devices.js [hunk 3]
```diff
@@ -53,13 +53,29 @@ class DeviceCommsHandler {
...
+            this.deviceLogHeartbeats[beat.id] = beat.timestamp
+        })
+
+        this.deviceLogHeartbeatInterval = setInterval(() => {
+            const now = Date.now()
+            for (const [key, value] of Object.entries(this.deviceLogHeartbeats)) {
+                if (now - value > 25000) {
+                    const parts = key.split(':')
+                    this.sendCommand(parts[0], parts[1], 'stopLog', '')
+                    this.app.log.info(`Disable device logging ${parts[1]}`)
```
Comment (knolleary) [inline forge/comms/devices.js:74]:
```suggestion
                    this.app.log.info(`Disable device logging ${parts[1]} in team ${parts[0]}`)
```

**Review by knolleary — APPROVED** [review]
Approved pending two small logging state changes

**Pull request merged**